### PR TITLE
fix: Wait for ThresholdSignature success before switching to NewKeysActivated

### PR DESCRIPTION
**Pull request by marcellorigotti**
# Pull Request

Closes: PRO-1062

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have updated documentation where appropriate.

## Summary

- Modified rotation to wait for ThresholdSignature success before switching to NewKeysActivated
- Modified tests to match the new behaviour

#### Non-Breaking changes

If this PR includes non-breaking changes, select the `non-breaking` label. On merge, CI will automatically cherry-pick the commit to a PR against the release branch.

### CODE REVIEW

**Comment by kylezs** [comment]
If you haven't already, I'd run a 3 node localnet and try to break this - running through some rotations, swaps through rotations etc. bouncer only does one rotation which isn't enough to catch a lot of the more likely small issues here

File: state-chain/traits/src/lib.rs
Comment (dandanlen) [inline state-chain/traits/src/lib.rs:200]: Doc comment would be nice.

Comment (dandanlen) [reply state-chain/traits/src/lib.rs:200]: Maybe rename `activate` -> `start_key_activation`.

File: state-chain/pallets/cf-broadcast/src/lib.rs
Comment (dandanlen) [inline state-chain/pallets/cf-broadcast/src/lib.rs:681]: nit: remove the leading `_` or simply inline the call.

**Review by dandanlen — COMMENTED** [review]
LGTM apart from some minor comments. 

I think it makes sense to await the approval of @ramizhasan111 before merging.

File: state-chain/traits/src/lib.rs
Comment (marcellorigotti) [reply state-chain/traits/src/lib.rs:203]: I added something, not sure how specific I have to be.
Is it ok like this?

File: state-chain/pallets/cf-threshold-signature/src/mock.rs
Comment (kylezs) [inline state-chain/pallets/cf-threshold-signature/src/mock.rs:203]: This suggests we don't have any unit tests that test the new code path? Since the new code path takes the path when this returns Some(). We should probably add a couple. I ran every code path in my brain to confirm this was the case. 

(I added a panic here and unit tests all pass)
```
						if Signature::<T, I>::get(tss_request_id) == AsyncResult::Void {
							panic!("check");
							T::VaultActivator::activate_key();
						};
```

Comment (marcellorigotti) [reply state-chain/pallets/cf-threshold-signature/src/mock.rs:203]: we have an integration test for it, I just tried adding the panic and it correctly fails the test.

I am referring to: integration-tests/src/authorities/authority_rotates_with_the_correct_sequence

Comment (kylezs) [reply state-chain/pallets/cf-threshold-signature/src/mock.rs:203]: My thinking is that having at least one unit test to isolate this particular flow means that when code changes and something like an integration test (which is harder to ensure is correct) is changed, nothing is missed, and it's caught in the tests

File: state-chain/pallets/cf-threshold-signature/src/key_rotator.rs
Comment (ramizhasan111) [inline state-chain/pallets/cf-threshold-signature/src/key_rotator.rs:150]: to keep the naming consistent with elsewhere maybe we should rename request_id to maybe_request_id and rename tss_request_id to request_id below.

**Review by ramizhasan111 — APPROVED** [review]

File: state-chain/pallets/cf-threshold-signature/src/key_rotator.rs
Comment (ramizhasan111) [reply state-chain/pallets/cf-threshold-signature/src/key_rotator.rs:163]: // After a tss ceremony completes, it is consumed and Void is left
						// behind At this point we are sure the ceremony existed and we
						// completed it we can activate the key
```suggestion
						// After the ceremony completes, the signature is consumed and Void is left
						// behind. At this point we are sure the ceremony existed and succeeded.
						//  We can activate the key
```

File: state-chain/pallets/cf-threshold-signature/src/lib.rs
Comment (marcellorigotti) [reply state-chain/pallets/cf-threshold-signature/src/lib.rs:144]: As discussed IRL, a migration is not necessary in this case, since we won't perform an upgrade during a rotation

**Comment by ramizhasan111** [comment]
Please dont merge this PR yet. It will have conflicts with the Arbitrum PR. I will merge this after merging the arbitrum PR.

**Comment by kylezs** [comment]
Discussed with @ramizhasan111 - merging this, as we're gonna keep the arb one open until we include some changes for the next release - including this one :)

**Pull request merged**
